### PR TITLE
implement GVS ID assignment

### PR DESCRIPTION
**Pull request by ahaessly**
Assign Ids to samples.
Update the sample info table.
Update import wdl to remove the generation of the sample_info.tsv files

### CODE REVIEW

File: scripts/variantstore/wdl/GvsAssignIds.wdl [hunk 1]
```diff
@@ -0,0 +1,168 @@
+version 1.0
+
+workflow GvsAssignIds {
+
+  input {
+    Array[String] external_sample_names
```
Comment (kcibul) [inline scripts/variantstore/wdl/GvsAssignIds.wdl:6]: Would this be more usable as a File that we then do a read_lines on?  Or have both options?

File: scripts/variantstore/wdl/GvsAssignIds.wdl [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+      if [ ~{has_service_account_file} = 'true' ]; then
+        gsutil cp ~{service_account_json} local.service_account.json
+        gcloud auth activate-service-account --key-file=local.service_account.json
+      fi
+
+      echo "project_id = ~{project_id}" > ~/.bigqueryrc
+
+      # create the lock table
+      bq --project_id=~{project_id} mk ~{dataset_name}.metadata_lock
```
Comment (kcibul) [inline scripts/variantstore/wdl/GvsAssignIds.wdl:82]: just checking, this script exits with an error of the metadata_lock table already exists?

Comment (kcibul) [reply scripts/variantstore/wdl/GvsAssignIds.wdl:82]: also -- maybe we should name this differently now that this isn't going into the "metadata" table?  sample_info_lock or sample_id_assignment_lock?

Comment (kcibul) [reply scripts/variantstore/wdl/GvsAssignIds.wdl:82]: also also… I'm surprised that this doesn't have to specify the schema when the table is created?

File: scripts/variantstore/wdl/GvsImportGenomes.wdl [hunk 3]
```diff
@@ -132,29 +126,11 @@ workflow GvsImportGenomes {
...
-    }
-  }
-
-  scatter (i in range(GetMaxTableId.max_table_id)) {
+  scatter (i in range(select_first([GetSampleIds.max_table_id, GetMaxTableIdLegacy.max_table_id]))) {
     call LoadTable as LoadPetTable {
     input:
       project_id = project_id,
-      table_id = i + 1,
+      table_id = i,
```
Comment (kcibul) [inline scripts/variantstore/wdl/GvsImportGenomes.wdl:133]: Won't `i` start at 0 here?

File: scripts/variantstore/wdl/GvsImportGenomes.wdl [hunk 4]
```diff
@@ -168,11 +144,11 @@ workflow GvsImportGenomes {
     }
   }
 
-  scatter (i in range(GetMaxTableId.max_table_id)) {
+  scatter (i in range(select_first([GetSampleIds.max_table_id, GetMaxTableIdLegacy.max_table_id])) {
     call LoadTable as LoadVetTable {
     input:
       project_id = project_id,
-      table_id = i + 1,
+      table_id = i,
```
Comment (kcibul) [inline scripts/variantstore/wdl/GvsImportGenomes.wdl:151]: same as above

File: scripts/variantstore/wdl/GvsImportGenomes.wdl [hunk 5]
```diff
@@ -43,32 +41,30 @@ workflow GvsImportGenomes {
       preemptible_tries = preemptible_tries
   }
 
-  call GetMaxTableId {
-    input:
-      sample_map = sample_map
+  if (defined(sample_map)) {
```
Comment (kcibul) [inline scripts/variantstore/wdl/GvsImportGenomes.wdl:44]: What do we need the legacy mode for?

**Review by kcibul — APPROVED** [review]
just a few comments, the id assignment part looks good!

File: scripts/variantstore/wdl/GvsAssignIds.wdl (hunk 2 above)
Comment (ahaessly) [reply scripts/variantstore/wdl/GvsAssignIds.wdl:82]: will explicitly check whether the table exists and exit with error if it does. 
renaming to sample_id_assignment_lock
adding schema when creating

File: scripts/variantstore/wdl/GvsImportGenomes.wdl (hunk 5 above)
Comment (ahaessly) [reply scripts/variantstore/wdl/GvsImportGenomes.wdl:44]: will leaving this in be helpful for anyone? I wasn't sure if people had tests that used a sample map file. Although, this won't update the sample_info table anymore, so maybe I should just remove it. If anyone else has an opinion, let me know.

**Pull request merged**